src/main/java/com/example/WatchNext/controller/AuthController.java

```
@@ -0,0 +1,91 @@
+                          PasswordEncoder encoder, SendEmail sendEmail, GeneratePassword generatePassword) {
+        this.authenticationManager = authenticationManager;
+        this.jwtUtils = jwtUtils;
+        this.userRepository = userRepository;
+        this.encoder = encoder;
+        this.sendEmail = sendEmail;
+        this.generatePassword = generatePassword;
+    }
+
+    @PostMapping("/signin")
+    public ResponseEntity<?> authenticateUser(@RequestBody LoginRequest loginRequest) {
+
+        Authentication authentication = authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(loginRequest.getUsername(), loginRequest.getPassword()));
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        String jwt = jwtUtils.generateJwtToken(authentication);
+
+        UserDetailsImpl userDetails = (UserDetailsImpl) authentication.getPrincipal();
+        Collection<? extends GrantedAuthority> roles = userDetails.getAuthorities();
+
+        return ResponseEntity.ok(new JwtResponse(jwt,
+                userDetails.getId(),
+                userDetails.getUsername(),
+                userDetails.getEmail(),
+                roles));
+    }
+
+    @PostMapping("/reset-password")
+    public ResponseEntity<?> registerUser(@RequestBody ResetPasswordRequest resetPasswordRequest) {
+
+        if (userRepository.existsByEmail(resetPasswordRequest.getEmail())) {
+            Users users = userRepository.findByEmail(resetPasswordRequest.getEmail());
+            String password = generatePassword.generateRandomPassword();
+            if (generatePassword.isPasswordValid(password)) {
+                users.setPassword(encoder.encode(password));
+                userRepository.save(users);
+                sendEmail.sendMail(users.getEmail(), "Reset password", password);
+            }
+            return ResponseEntity.ok(new MessageResponse(
```

Well....if the generated password is not valid the email will not be sent, but you are going to send the response "was successful" message back....I think this is a **silent fail**.